unit_test/main_tests/main_platform_test.py

```diff
@@ -112,6 +114,7 @@ def test_archs_platform_specific(platform, intercepted_build_args, monkeypatch):
         assert build_options.architectures == {Architecture.x86_64}
 
 
+@pytest.mark.allow_empty('windows')
```

Comment:
Why don't we just skip Windows, then? Or fix the `machine`?

Reply:
I originally had a fixture for this. Then I realized there were a lot of these, so I added the marker. But then I fixed most of them by making the arch realistic for the testing, so it probably could be a fixture again.

Reply:
I've reworked it into a fixture, and fixed several of the tests to no longer need it (actually tests more thoroughly now :) ), and I've added an `--allow-empty` option that lets this pass in a shell-independent way. I think it will help the transition, especially if it turns out someone does like using `SKIP=*` or similar as an API. If we had never let empty builds pass, I don't think it would be needed.

Reply:
I would be okay to leave it an undocumented option and just see if users end up needed it. If enough do, we could document it.